sklearn/preprocessing/data.py

```diff
@@ -656,8 +662,9 @@ def partial_fit(self, X, y=None):
                     self.var_ = None
 
             self.mean_, self.var_, self.n_samples_seen_ = \
-                _incremental_mean_and_var(X, self.mean_, self.var_,
-                                          self.n_samples_seen_)
+                _incremental_mean_and_var(
+                    X, self.mean_, self.var_,
+                    self.n_samples_seen_ * np.ones(X.shape[1]))
```

Comment:
I don't get it. From the changes, I though that `self.n_samples_seen_` should always be an array.

> And yet, for backwards compatibility, we need to report only a scalar in cases that are not affected by this PR (i.e. where there are no NaNs, or perhaps where n_samples_seen_ is constant even if there were NaNs).

I thought that it would be easier to change only array from now on. Only `incremental_pca` is affected apart of the `StandardScaler` and those functions are private so the end-user should not care.

Reply:
`n_samples_seen_` is not private IMO

Reply:
@jnothman @glemaitre Sorry for being inactive for the past week. Shall I keep `self.n_sample_seen_` a vector or scaler?
PS: As of now, `self.n_sample_seen_` is a vector both in `StandardScaler` and `incremental_pca`

Reply:
IMO it would be best for backwards compatibility to keep a scalar in the case when there are no NaNs or - for simplicity - in the case when all n_samples_seen are equal

Reply:
Fair enough

Reply:
@jnothman @glemaitre I'll make them generalised i.e. if all n_samples_seen are equal, it will return a scalar instead of a vector.